TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs

```diff
@@ -64,32 +53,17 @@ public IEnumerable<SecurityToken> SecurityTokens
         {
             get
             {
-                RetrieveMetadata();
-                _synclock.EnterReadLock();
-                try
-                {
-                    return _tokens;
-                }
-                finally
-                {
-                    _synclock.ExitReadLock();
-                }
+                return _tokens;
             }
         }
 
-        private void RetrieveMetadata()
+        private async Task<OpenIdConnectConfiguration> RetrieveMetadata()
         {
-            _synclock.EnterWriteLock();
-            try
-            {
-                OpenIdConnectConfiguration config = Task.Run(_configManager.GetConfigurationAsync).Result;
-                _issuer = config.Issuer;
-                _tokens = config.SigningTokens;
-            }
-            finally
-            {
-                _synclock.ExitWriteLock();
-            }
+            OpenIdConnectConfiguration config = await _configManager.GetConfigurationAsync();
+            _issuer = config.Issuer;
```

Comment:
We don't need to cache into local variables. ConfigurationManager is designed to have controlled refresh interval.
GetConfigurationAsync() uses the property AutomaticRefreshInterval to determine if it should query the discovery endpoint.
This is important when keys rollover.